python/lsst/ts/cbp/component.py

```diff
@@ -165,250 +186,233 @@ async def disconnect(self):
-        self.encoder_status.MASK_SELECT = bool(await self.send_command("ACstat=?"))
-        self.encoder_status.MASK_ROTATE = bool(await self.send_command("ADstat=?"))
-        self.encoder_status.FOCUS = bool(await self.send_command("AEstat=?"))
+        panic = bool(int(await self.send_command("wdpanic=?")))
+        azimuth = bool(int(await self.send_command("AAstat=?")))
+        elevation = bool(int(await self.send_command("ABstat=?")))
+        mask = bool(int(await self.send_command("ACstat=?")))
+        mask_rotation = bool(int(await self.send_command("ADstat=?")))
+        focus = bool(int(await self.send_command("AEstat=?")))
+        self.csc.tel_status.set_put(
+            panic=panic,
+            azimuth=azimuth,
+            elevation=elevation,
+            mask=mask,
+            mask_rotation=mask_rotation,
+            focus=focus,
+        )
 
     async def get_cbp_telemetry(self):
-        """Gets the position data of the CBP.
-
-        Returns
-        -------
-        None
+        """Get the position data of the CBP.
 
         """
-        await self.get_altitude()
+        await self.get_elevation()
         await self.get_azimuth()
         await self.get_focus()
         await self.get_mask()
-        await self.get_mask_rotation()
 
     def configure(self, config):
+        """Configure the CBP.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+        """
         self.host = config.address
         self.port = config.port
-        self.masks.__dict__.get("1").name = config.mask1["name"]
-        self.masks.__dict__.get("1").rotation = config.mask1["rotation"]
-        self.masks.__dict__.get("2").name = config.mask2["name"]
-        self.masks.__dict__.get("2").rotation = config.mask2["rotation"]
-        self.masks.__dict__.get("3").name = config.mask3["name"]
-        self.masks.__dict__.get("3").rotation = config.mask3["rotation"]
-        self.masks.__dict__.get("4").name = config.mask4["name"]
-        self.masks.__dict__.get("4").rotation = config.mask4["rotation"]
-        self.masks.__dict__.get("5").name = config.mask5["name"]
-        self.masks.__dict__.get("5").rotation = config.mask5["rotation"]
-
-    async def publish(self):
-        """This updates the attributes within the component.
-
-        Returns
-        -------
-        None
+        self.masks.get("1").name = config.mask1["name"]
+        self.masks.get("1").rotation = config.mask1["rotation"]
+        self.masks.get("2").name = config.mask2["name"]
+        self.masks.get("2").rotation = config.mask2["rotation"]
+        self.masks.get("3").name = config.mask3["name"]
+        self.masks.get("3").rotation = config.mask3["rotation"]
+        self.masks.get("4").name = config.mask4["name"]
+        self.masks.get("4").rotation = config.mask4["rotation"]
+        self.masks.get("5").name = config.mask5["name"]
+        self.masks.get("5").rotation = config.mask5["rotation"]
```

Comment:
I suggest replacing `.get("n")` with `["n"]`. This improves the error reporting if the key is missing. The current code will complain that "None has no attribute name" and not tell you wish key is the problem. With `["n"]` Python will complain that key "n" is missing and not try to use None as a mask.